src/ocean/global.F90

```diff
@@ -251,53 +248,85 @@ integer function find_row2(i,j,k,XX)
 
   end function find_row2
 
-  !! this is a part of the subroutine grid from grid.f. We need it to
-  !! initialize the vectors x,y,z...
-  !! It should be kept consistent with that subroutine
-  SUBROUTINE g_grid
-    use m_par
+  subroutine get_x(asize,ptr) bind(C,name="get_global_x")
+    use, intrinsic :: iso_c_binding
+    implicit none
+    integer(c_int), intent(out) :: asize
+    type(c_ptr), intent(out) :: ptr
+
+    asize = size(x)
+    ptr = c_loc(x(1))
```

Comment:
Why does the memory have to live on the fortran side?

Reply:
I wasn't sure how having the Fortran side use the C++ pointer would interact with the arrays that index from something other than 0. So this was the quickest way to get things work, but I can probably change it to have C++ own the poitners.

Reply:
It's something we also do with the matrices I think. But if you do all of this from the C++ side you can just pass them to `global_init` (or something like that) in one go without all of these functions.